cmd/abapEnvironmentAssembly.go

```diff
@@ -0,0 +1,598 @@
+package cmd
+
+import (
+	"bytes"
+	"path"
+	"path/filepath"
+	"sort"
+	"time"
+
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http/cookiejar"
+
+	"github.com/SAP/jenkins-library/pkg/abaputils"
+	"github.com/SAP/jenkins-library/pkg/command"
+	piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"github.com/pkg/errors"
+)
+
+func abapEnvironmentAssembly(config abapEnvironmentAssemblyOptions, telemetryData *telemetry.CustomData, commonPipelineEnvironment *abapEnvironmentAssemblyCommonPipelineEnvironment) {
+	// for command execution use Command
+	c := command.Command{}
+	// reroute command output to logging framework
+	c.Stdout(log.Writer())
+	c.Stderr(log.Writer())
+
+	// for http calls import  piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	// and use a  &piperhttp.Client{} in a custom system
+	// Example: step checkmarxExecuteScan.go
+
+	// error situations should stop execution through log.Entry().Fatal() call which leads to an os.Exit(1) in the end
+	err := runAbapEnvironmentAssembly(&config, telemetryData, &c, commonPipelineEnvironment)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func runAbapEnvironmentAssembly(config *abapEnvironmentAssemblyOptions, telemetryData *telemetry.CustomData, command command.ExecRunner, commonPipelineEnvironment *abapEnvironmentAssemblyCommonPipelineEnvironment) error {
+	return runAssembly(config)
+}
+
+// *******************************************************************************************************************************
+// ********************************************************** Step logic *********************************************************
+// *******************************************************************************************************************************
+func runAssembly(config *abapEnvironmentAssemblyOptions) error {
```

Comment:
There seems to be no benefit in having those two functions. It seems that "runAssembly" can be omitted.
I guess "runAbapEnvironmentAssembly" was generated. Nevertheless, you can change the signature if you want to.